ReactAndroid/src/main/java/com/facebook/react/devsupport/MultipartStreamReader.java

```diff
@@ -98,6 +119,20 @@ public boolean readAllParts(ChunkCallback callback) throws IOException {
 
       if (indexOfDelimiter == -1) {
         bytesSeen = content.size();
+
+        if (currentHeaders == null) {
+          long indexOfHeaders = content.indexOf(headersDelimiter, searchStart);
+          if (indexOfHeaders >= 0) {
+            mSource.read(content, indexOfHeaders);
+            Buffer headers = new Buffer();
+            content.copyTo(headers, searchStart, indexOfHeaders - searchStart);
+            currentHeadersLength = headers.size();
+            currentHeaders = parseHeaders(headers);
+          }
+        } else {
+          emitProgress(currentHeaders, content.size() - currentHeadersLength, false, progressCallback);
```

Comment:
Instead of currentHeadersLength (assuming we want to calculate the size of the response body downloaded so far), don't we want to track the starting index of the response body?

Reply:
Yea there might be an error here, we don't take into consideration the length of the headers delimiter. I could keep currentHeadersLength and do `content.size() - currentHeadersLength - headersDelimiter .size()` or change currentHeadersLength for something like contentStart.